src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -138,6 +141,22 @@ static BoundNode unwrapListElement(BoundCollectionExpression node, BoundNode ele
             }
         }
 
+        private bool IsSingleReadOnlySpanSpread(BoundCollectionExpression node, [NotNullWhen(true)] out BoundExpression? spreadExpression)
+        {
+            spreadExpression = null;
+
+            if (node.Elements is
+                [
+                    BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr, IteratorBody: BoundExpressionStatement { Expression: not BoundConversion { ConversionKind: not ConversionKind.Identity } } }
```

Comment:
>IteratorBody: BoundExpressionStatement { Expression: not BoundConversion { ConversionKind: not ConversionKind.Identity } } [](http://example.com/codeflow?start=116&length=122)

I must admit, I find this pattern difficult to read with `not ... { ... not ... }` and difficult to convince myself that it is correct.

An alternative might be to check for expected values directly, something like:
```
if (node.Elements is
    [
        BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr, IteratorBody : BoundExpressionStatement { Expression: var item } }
    ] &&
    item is BoundValuePlaceholder or BoundConversion { Conversion.Kind: ConversionKind.Identity, Operand: BoundValuePlaceholder { } } &&
    spreadType.OriginalDefinition == (object)_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T))
```

But that alternative is also complicated. And both approaches suffer from trying to unpack the `BoundExpression` that was created in binding.

It feels like the original approach, before commit 9, which checked the type of the spread element, was more obviously correct.

Would it make sense to simply check there is an identity conversion from the spread element expression to the builder method parameter?
```
if (node.Elements is
    [BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol spreadType } expr }] &&
    Conversions.HasIdentityConversion(node.CollectionBuilderMethod.Parameters[0].Type, spreadType))
{
```